testing/test_pytest_html.py

```diff
@@ -14,20 +14,15 @@
 pytest_plugins = ("pytester",)
 
 
-def handle_tr_writer_deprecation():
-    # Remove this function when they've fixed
+def remove_deprecation_from_recwarn(recwarn):
+    # TODO: Temporary hack until they fix
     # https://github.com/pytest-dev/pytest/issues/6936
-    import warnings
-    from _pytest.warnings import _setoption
-
-    arg = "ignore:TerminalReporter.writer:pytest.PytestDeprecationWarning"
-    _setoption(warnings, arg)
+    for index, item in enumerate(recwarn):
+        if "TerminalReporter.writer" in repr(item.message):
+            del recwarn.list[index]
```

Comment:
I don't think this is really safe if there are multiple instances in the list, but in practice that likely does not matter here.

(if you delete index 0 in a list of length 2, enumerate will never even product the second item).

Reply:
Aha, is it better to store the relevant indexes and then after the iteration is complete delete them? @dhalperi

Reply:
see suggestion - basically, for me, not modifying recwarn seems easier than tracking indices, etc.